inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -268,6 +269,40 @@ void HandleMultipleActivationsForTheLayerPass::run() {
     }
 }
 
+void ForbidActivationFusingPass::run() {
+    for (auto& l : *pLayers) {
+        if (l->insData.empty()) continue;
+        auto prevLayer = CNNNetPrevLayerSkipCertain(l, 0, [](CNNLayerPtr ptr) {
+            return LayerInfo(ptr).isNonFunctional();
+            });
+        if (LayerInfo(l).isActivation()) {
```

Comment:
I guess it is enough to detect if it is activation layer.
Then for activation layer check if previous layer has more than 1 layer connected. If it is the case, then the diagonal layer needs to be added.
I don't think you need to skip any layers when you search for all directly connected layers to the previous one. If you have other opinion, please add in comment what cases need to be skipped and why.

Reply:
It is done that way. But I can try to simplify this by reordering and using CNNNetPrevLayer() instead CNNNetPrevLayerSkipCertain().

Reply:
It's simplified now.